x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/timeline/parser.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty } from 'lodash';
+import { Plugin } from '@elastic/eui/node_modules/unified';
+import { RemarkTokenizer } from '@elastic/eui';
+import { ID, PREFIX, OLD_PREFIX } from './constants';
+import { TimelineConfiguration } from './types';
+
+const START_POS = PREFIX.length;
+
+const requiredFields = ['id', 'title'];
+
+const validateConfiguration = (configuration: TimelineConfiguration) => {
+  requiredFields.forEach((field) => {
+    if (isEmpty(configuration[field])) {
+      throw new Error(`${field} is missing.`);
+    }
+  });
+};
+
+export const TimelineParser: Plugin = function () {
+  const Parser = this.Parser;
+  const tokenizers = Parser.prototype.inlineTokenizers;
+  const methods = Parser.prototype.inlineMethods;
+  let oldFormat = false;
+
+  const parseOldFormat: RemarkTokenizer = function (eat, value, silent) {
+    let index = 0;
+    const nextChar = value[index];
+
+    if (nextChar !== '[') {
+      return false;
+    }
+
+    if (silent) {
+      return true;
+    }
+
+    function readArg(open: string, close: string) {
```

Comment:
this looks like it was fun 😜

Reply:
Hahaha! Fortunately, the EUI team did it for us :stuck_out_tongue: